TrackingTools/TrajectoryFiltering/interface/MinHitsTrajectoryFilter.h

```diff
@@ -49,16 +49,19 @@ class MinHitsTrajectoryFilter final : public TrajectoryFilter {
   bool QF(const T& traj) const {
     int seedPenalty = (2 == traj.seedNHits()) ? theSeedPairPenalty : 0;  // increase by one if seed-doublet...
     bool passed = false;
-    double pt = traj.lastMeasurement().updatedState().freeTrajectoryState()->momentum().perp();
-    double pz = traj.lastMeasurement().updatedState().freeTrajectoryState()->momentum().z();
-    double sinhTrajEta2 = (pz * pz) / (pt * pt);
-    double myEtaSwitch = sinh(theHighEtaSwitch);
-    if (sinhTrajEta2 < (myEtaSwitch * myEtaSwitch)) {
-      if (traj.foundHits() >= theMinHits + seedPenalty)
-        passed = true;
-    } else {  //absTrajEta>theHighEtaSwitch, so apply relaxed cuts
-      if (traj.foundHits() >= theMinHitsAtHighEta + seedPenalty)
-        passed = true;
+
+    if (!traj.empty()) {
+      auto pt = traj.lastMeasurement().updatedState().freeTrajectoryState()->momentum().perp();
```

Comment:
this should be 
auto pt2 = traj.lastMeasurement().updatedState().freeTrajectoryState()->momentum().perp2();
and later of course auto 
sinhTrajEta2 = (pz * pz) /pt2;